packages/scene-composer/src/components/panels/scene-components/AnimationComponentEditor.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormField, Button, SpaceBetween, Autosuggest } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IAnimationComponentInternal, ISceneNodeInternal } from '../../../store';
+import { getAnimationObjectByName } from '../../three-fiber/AnimationComponent/AnimationComponent';
+
+export type AnimationComponentEditorProps = {
+  node: ISceneNodeInternal;
+  component: IAnimationComponentInternal;
+  onUpdate;
+};
+export const AnimationComponentEditor: React.FC<AnimationComponentEditorProps> = ({
+  node,
+  component: animationComponent,
+  onUpdate,
+}: AnimationComponentEditorProps) => {
+  const { formatMessage } = useIntl();
+
+  const [panelnumber, setPanelNumber] = React.useState(0);
+  const animationList = getAnimationObjectByName(animationComponent.uri);
+  console.log(animationList)
+  const [selectedAnimations, setSelectedAnimations] = React.useState<string[]>();
+
+  const animationOptions = useMemo(
+    () =>
+      animationList?.animations.map((animation) => ({
+        value: animation.name,
+      })),
+    [animationList, node],
+  );
+
+  const updateAnimations = useCallback(
+    (animation, i) => {
+      if (selectedAnimations) {
+        const currentAnimations = [...selectedAnimations];
+        currentAnimations[i] = animation;
+        setSelectedAnimations(currentAnimations);
+      }
+    },
+    [selectedAnimations],
+  );
+
+  const removeAnimation = useCallback(
+    (i) => {
+      if (selectedAnimations) {
+        const currentAnimations = [...selectedAnimations];
+        currentAnimations.splice(i, 1);
+        setSelectedAnimations(currentAnimations);
+      }
+    },
+    [selectedAnimations],
+  );
+
+  useEffect(() => {
+    const currentAnimations: string[] = [...animationComponent.currentAnimations] || [];
+    if (animationComponent.selector) {
+      if (animationComponent.selector > panelnumber) {
+        for (let i = panelnumber; i < animationComponent.selector; i++) {
+          currentAnimations.push('');
+          setPanelNumber(panelnumber + 1);
+        }
+      }
+    }
+    setSelectedAnimations(currentAnimations);
+  }, [node.ref, animationComponent.selector]);
+
+  useEffect(() => {
+    if (selectedAnimations) {
+      onUpdate(selectedAnimations, node.ref, animationComponent);
+    }
+  }, [selectedAnimations]);
```

Comment:
This is probably causing an extra event when you change node.  (node changes, selected animations changes resulting in us saving to the component the values it already has). Lets make an Asana task to clean it up as I know it's being a bit tricky right now and we want this baseline PR in.